sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -183,6 +183,14 @@ public virtual bool IsClosed
             private set => _closed = value;
         }
 
+        /// <summary>
+        /// Gets diagnostics information for the processor.
+        /// </summary>
+        public virtual ServiceBusProcessorDiagnostics Diagnostics =>
+            _diagnostics ?? throw new InvalidOperationException("Diagnostics are not enabled. To enable diagnostics, set the EnableDiagnostics property on the ServiceBusProcessorOptions.");
```

Comment:
The FDG says to avoid this in favor of a method. I wonder if we can get away with it here.